src/hist_smooth/base_smoothing.py

```diff
@@ -1,9 +1,32 @@
 import numpy as np
 from numpy.lib.stride_tricks import sliding_window_view
-
-
-def moving_median_odd(data, window, endrule='median'):
-    """Moving median with odd window size."""
+from typing import Callable
+
+
+def moving_median_odd(data: np.ndarray | list[float], window: int, endrule: str = "median") -> np.ndarray:
+    """
+    Apply a moving median filter with an odd window size.
+
+    Parameters
+    ----------
+    data : array_like of float
+        Input 1D sequence of numeric values.
+    window : int
+        Size of the moving window (must be odd).
+    endrule : {"median", "keep"}, optional
+        Rule for handling boundaries in median smoothing.
+        Default is "median".
+
+    Returns
+    -------
+    np.ndarray
+        Smoothed array with the same length as the input.
+
+    Raises
+    ------
+    ValueError
+        If `endrule` is not one of {"median", "keep"}.

```

Comment:
```suggestion
```

In general what errors a function can raise aren't things that are documented in docstrings. These should be things that are communicated to the user through API design and checks.